


By evaluating this approach in regards to genotype accuracy and power to detect associations using meta-imputed data,

Notably, genotypes that reflect the true, underlying genotypic state
from a set of candidate variants.
For example, in comparisons between meta-imputed genotypes and those imputed from panels of distant geographic
the AFR and ASN panels in scenario B, for which lower accuracy levels may be expected due to the ethnic background of the study sample, which was derived from a sample of mainly European ancestry.

Variants of low risk effect size are generally less distinctive from stochastic noise in the data, where such observations may be attributed to accumulation of false positive signals.

As currently implemented, the algorithm attempts to by harness most of the information available across multiple reference datasets.

By conducting accuracy analyses and association testing in extensive case-control simulations, we have shown that our method, meta-imputation, has the potential to further increase power in \gls{gwa} studies.

output genotypes
dependent on merge operation
MSS rearranges data imputed from different sources, such that the resulting genotype matrix represents a mosaic of estimated genotypes, where posterior probabilities are unaltered.
On the other hand, weighted linear combination (WLC) averages posterior probabilities over candidate genotypes.
which is why the tuple values cannot safely be interpreted as posterior probabilities.
Therefore, it is appropriate to report meta-imputed genotypes as integers rounded to the most likely genotypic state when using WLC.

Although meta-imputation is unlikely to increase accuracy and power further than possible
what can be expected through imputation from a large, canonical reference
in contrast to \gls{hrc}-based imputation, it has the benefit that researchers can choose which reference data to combine.
For example, to date most research has been conducted on samples collected from European backgrounds, whereas other populations are currently under-sampled.
Also, ungenotyped sites in highly recombinant chromosomal regions, e.g. the Major Histocompatibility Complex (MHC) region, are difficult to estimate \citep{Chen:2013jx}, because imputation depends on matching \gls{ld} patterns between a reference and the sample. By combining genotypes imputed from targeted sequence data (e.g. \correct{[REF]}) with those from more general panels, the number of variants can be extended to potentially uncover more disease-contributing factors.
Hence, future \gls{gwas} may benefit from an approach such as meta-imputation.


future directions:
- testing meta-imputation on HRC imputed genotypes with other e.g. non-European reference imputations
- testing other meta-imputation apporaches
- performing meta-imputation with different layers of quality control, e.g. retaining variants only if present in at least to panels


Future gwas are likely to benefit from such a resource

faces logistic challenges as well as legal restrictions for data sharing, storage, and dissemination of personal information of study participants.

For example, \gls{hrc} data currently contains individuals of European ancestry,
which is likely to impose certain limitations for imputation into non-European samples.

Generally, computation time of imputation increases with both the number of samples and the number of markers of the reference panel \cite{Howie:2012ks}.

Although \gls{hrc} data has the potential to become one of the primary resources for imputation in future \gls{gwas},






% \begin{align*}
% 	\textit{Rare variants}           & \quad \rightarrow \quad {\text{MAF} \in \left[ 0.00, 0.01\right]} \\
% 	\textit{Low-frequency variants}  & \quad \rightarrow \quad {\text{MAF} \in \left( 0.01, 0.05\right]} \\
% 	\textit{Common variants}         & \quad \rightarrow \quad {\text{MAF} \in \left( 0.05, 0.50\right]}
% \end{align*}


%Mean genotype accuracy, $\bar r^2$, was then calculated per bin.
% The relative accuracy difference between \n{2} imputed datasets was expressed as $\delta\bar r^2$, which was obtained on the intersected set of variants and calculated as the averaged difference between $r^2$ values by variant.
