%
% Abbreviations
%

% 0-9
\newacronym{1kg}{1000G}{1000~Genomes Project}
% A
\newacronym{afs}{AFS}{allele frequency spectrum}
\newacronym{arg}{ARG}{ancestral recombination graph}
% B

% C
\newacronym[plural=cM,firstplural=centiMorgans~(cM)]{cM}{cM}{centiMorgan}
\newacronym{cdf}{CDF}{cumulative distribution function}
\newacronym{ccf}{CCF}{cumulative coalescent function}
% D
\newacronym{dna}{DNA}{deoxyribonucleic acid}
\newacronym{dgt}{DGT}{discordant genotype test}
% E
\newacronym{em}{EM}{Expectation-Maximization}
\newacronym{ebi}{EBI}{European Bioinformatics Institute}
% F
\newacronym{fgt}{FGT}{four-gamete test}
\newacronym{fpr}{FPR}{false positive rate}
\newacronym{fnr}{FNR}{false negative rate}
% G
\newacronym{got2d}{GoT2D}{Genetics of Type~2 Diabetes Project}
\newacronym{gwa}{GWA}{genome-wide association}
\newacronym{gwas}{GWAS}{genome-wide association studies}
\newacronym{gonl}{GoNL}{Genome of the Netherlands Project}
\newacronym{gof}{GoF}{gain-of-function}
% H
\newacronym{hrc}{HRC}{Haplotype Reference Consortium}
\newacronym{hwe}{HWE}{Hardy-Weinberg equilibrium}
\newacronym{hapmap}{HapMap}{International HapMap Project}
\newacronym{hmm}{HMM}{Hidden Markov Model}
\newacronym{hgp}{HGP}{Human Genome Project}
% I
\newacronym{ism}{ISM}{infinite sites model}
\newacronym{ibd}{IBD}{identity by descent}
\newacronym{ibs}{IBS}{identity by state}
\newacronym{ipg}{IPG}{Illumina Platinum Genomes Project}
\newacronym{iid}{\emph{i.i.d.}}{independent and identically distributed}

% J

% K
\newacronym[plural=Kb,firstplural=kilobases~(Kb)]{Kb}{Kb}{kilobase}
% L
\newacronym{ld}{LD}{linkage disequilibrium}
\newacronym{le}{LE}{linkage equilibrium}
\newacronym{lrp}{LRP}{long range phasing}
\newacronym{lod}{LOD}{logarithm of odds}
\newacronym{lr}{LR}{likelihood ratio}
\newacronym{lof}{LoF}{loss-of-function}
% M
\newacronym{maf}{MAF}{minor allele frequency}
\newacronym[plural=Mb,firstplural=Megabases~(Mb)]{Mb}{Mb}{Megabase}
\newacronym{mrca}{MRCA}{most recent common ancestor}
% N
\newacronym{ngs}{NGS}{next-generation sequencing}
\newacronym{nhgri}{NHGRI}{National Human Genome Research Institute}
% O
\newacronym{orcades}{ORCADES}{Orkney Complex Disease Study}
\newacronym{ooa}{OOA}{Out-of-Africa model}
\newacronym{or}{OR}{odds ratio}
% P
\newacronym{pca}{PCA}{principle component analysis}
\newacronym{pdf}{PDF}{probability density function}
\newacronym{pmf}{PMF}{probability mass function}
\newacronym{pcr}{PCR}{polymerase chain reaction}
\newacronym{pbwt}{PBWT}{positional Burrows-Wheeler transform}
\newacronym{pcoa}{PCoA}{principal coordinates analysis}
\newacronym{psmc}{PSMC}{pairwise sequentially Markovian coalescent}

% Q
\newacronym{qc}{QC}{quality control}
% R
\newacronym{rmsle}{RMSLE}{root mean squared logarithmic error}
\newacronym{rna}{RNA}{ribonucleic acid}
\newacronym{rr}{RR}{relative risk}
\newacronym{rflp}{RFLP}{restriction fragment length polymorphism}
% S
\newacronym{sfs}{SFS}{site frequency spectrum}
\newacronym[plural=SNPs,firstplural=single-nucleotide polymorphisms~(SNP)]{snp}{SNP}{single-nucleotide polymorphism}
\newacronym{sisu}{SISu}{Sequencing Initiative Suomi Project}
\newacronym{se}{SE}{standard error}
\newacronym{smc}{SMC}{sequentially Markov coalescent}

% T
\newacronym{t2d}{T2D}{type 2 diabetes}
\newacronym{tmrca}{T\textsubscript{MRCA}}{time to the most recent common ancestor}
%\newacronym{tmrca}{T\textsubscript{MRCA}}{time to the most recent common ancestor}
% U
\newacronym{uk10k}{UK10K}{UK10K Genome Sequencing Project}
\newacronym{ukb}{UKB}{UK\,Biobank}
% V
\newacronym{vcf}{VCF}{Variant Call Format}
\newacronym{vep}{VEP}{Variant Effect Predictor}
% W
\newacronym{wgs}{WGS}{whole-genome sequencing}
\newacronym{wes}{WES}{whole-exome sequencing}
% X

% Y

% Z

%
\newacronym{lgc}{{\ensuremath{\lambda\text{GC}}}}{genomic control inflation factor}
