% !TEX root = ../../main.tex


\begin{table}[!htb]
\Caption{Accuracy of detected breakpoints per $\textbf{f}_{k}$~category}
{The accuracy of detected IBD breakpoints was measured using the squared Pearson correlation coefficient, $r^2$, and the \gls{rmsle} in relation to the true IBD segments determined from simulation records; measured in terms of the distance between breakpoint site and the corresponding focal position per segment.
The analysis included of \n{317020} target sites around which IBD was detected in \cref{app:fgt_h,,app:fgt_p,,app:dgt}.
In each, accuracy was computed after data were reduced to identical sets of unique IBD segments (${n=\num{2978220}}$).
The table specifies the allele frequency~(\%) corresponding to each \fk{}~category, as well as the number of target sites identified.}
{tab:stat_breakpoints}
\centering
\begin{threeparttable}
\begin{tabular}{ccS[table-format=6.1]*6{S[table-format=1.3]}}
\toprule
\fk{} & Freq.~{\%} & \multicolumn{1}{c}{Targets} &
\multicolumn{3}{c}{$r^2$} &
\multicolumn{3}{c}{$\rmsle$} \\
\cmidrule(lr){4-6}
\cmidrule(lr){7-9}
 & & &
 \multicolumn{1}{c}{\textsc{fgt}$^{\ast}$} &
 \multicolumn{1}{c}{\textsc{fgt}$^{\ast\ast}$} &
 \multicolumn{1}{c}{\textsc{dgt}$^{\dagger}$} &
 \multicolumn{1}{c}{\textsc{fgt}$^{\ast}$} &
 \multicolumn{1}{c}{\textsc{fgt}$^{\ast\ast}$} &
 \multicolumn{1}{c}{\textsc{dgt}$^{\dagger}$} \\
\midrule
2  & 0.04 & 76515 & 0.99764 & 0.89458 & 0.99520 & 0.21948 & 0.59833 & 0.31710 \\
3  & 0.06 & 46138 & 0.98869 & 0.95705 & 0.97784 & 0.24316 & 0.51595 & 0.35887 \\
4  & 0.08 & 31658 & 0.96294 & 0.95872 & 0.94097 & 0.25555 & 0.46282 & 0.37881 \\
5  & 0.10 & 23581 & 0.97480 & 0.96275 & 0.92906 & 0.27554 & 0.42873 & 0.40830 \\
6  & 0.12 & 19241 & 0.95448 & 0.93766 & 0.90396 & 0.28086 & 0.40886 & 0.42082 \\
7  & 0.14 & 15869 & 0.95501 & 0.94390 & 0.89159 & 0.29827 & 0.40264 & 0.44651 \\
8  & 0.16 & 13175 & 0.89794 & 0.91758 & 0.81310 & 0.31984 & 0.39805 & 0.46935 \\
9  & 0.18 & 10966 & 0.93158 & 0.92740 & 0.82685 & 0.31402 & 0.37511 & 0.46718 \\
10 & 0.20 & 11142 & 0.87942 & 0.88681 & 0.77257 & 0.33200 & 0.38713 & 0.49355 \\
11 & 0.22 &  9392 & 0.89481 & 0.89213 & 0.75844 & 0.34414 & 0.40133 & 0.51339 \\
12 & 0.24 &  7751 & 0.83526 & 0.84776 & 0.73282 & 0.35792 & 0.39763 & 0.52569 \\
13 & 0.26 &  6933 & 0.84208 & 0.83534 & 0.72078 & 0.36065 & 0.40463 & 0.53176 \\
14 & 0.28 &  5767 & 0.81623 & 0.81622 & 0.67937 & 0.36694 & 0.39075 & 0.54006 \\
15 & 0.30 &  5062 & 0.87140 & 0.86023 & 0.71246 & 0.38095 & 0.40601 & 0.55560 \\
16 & 0.32 &  4711 & 0.83929 & 0.82998 & 0.70121 & 0.37271 & 0.39465 & 0.54617 \\
17 & 0.34 &  4210 & 0.82865 & 0.83173 & 0.68057 & 0.38680 & 0.41049 & 0.56620 \\
18 & 0.36 &  3913 & 0.81267 & 0.83171 & 0.67040 & 0.37969 & 0.39699 & 0.56071 \\
19 & 0.38 &  3684 & 0.80141 & 0.79804 & 0.64162 & 0.38074 & 0.40128 & 0.56564 \\
20 & 0.40 &  3214 & 0.83058 & 0.83677 & 0.68478 & 0.40090 & 0.41599 & 0.58737 \\
21 & 0.42 &  3333 & 0.77276 & 0.77758 & 0.60320 & 0.39881 & 0.41259 & 0.58359 \\
22 & 0.44 &  2863 & 0.75260 & 0.79505 & 0.57059 & 0.39876 & 0.40632 & 0.58646 \\
23 & 0.46 &  2595 & 0.73156 & 0.74493 & 0.59629 & 0.41417 & 0.42509 & 0.59901 \\
24 & 0.48 &  2653 & 0.78359 & 0.77969 & 0.58121 & 0.39649 & 0.40641 & 0.58343 \\
25 & 0.50 &  2654 & 0.70052 & 0.72997 & 0.56030 & 0.39960 & 0.40764 & 0.58521 \\
\bottomrule
\end{tabular}
\begin{tablenotes}\footnotesize
	\item[$\ast$] ~ \cref{app:fgt_h}, using the \gls{fgt} with true haplotypes
	\item[$\ast\ast$] ~ \cref{app:fgt_p}, using the \gls{fgt} with phased haplotypes
	\item[$\dagger$] ~ \cref{app:dgt}, using the \gls{dgt} with genotype data
\end{tablenotes}
\end{threeparttable}
\end{table}

% 21803
% 36501
% 49533
% 59157
% 69271
% 89298
% 86317
% 98192
% 106765
% 118180
% 121107
% 123542
% 132232
% 139403
% 146324
% 150165
% 147554
% 149660
% 162293
% 176537
% 187870
% 195883
% 162501
% 94297

% 18795
% 34042
% 47338
% 59683
% 69487
% 81376
% 91580
% 98374
% 105827
% 116829
% 123145
% 127134
% 130059
% 145055
% 147945
% 151803
% 157849
% 160246
% 164855
% 183665
% 177742
% 181959
% 195948
% 207484
