



meiotic recombination breaks down
and mutations
erosion of ancestral identity over time
but also
and hence the ability to adapt to changing environments

\citet{cotterman1940calculus} used the term \emph{derivative}.
\Gls{ibd} \citet{malecot1948mathematics}
\citep{Thompson:2013cj}
recent coancestry
as older regions diversify by accumulating mutations

Rare variants

pairwise relationship between individuals



The probability that at a random site the break condition is satisfied
4 possible allelic state configurations
4 gametes
${4^4=256}$
${3^2=9}$ possible genotypic state configurations
2 individuals
${9^2=81}$




%
%
%
%
%
%
%
%
%
%
%
%


\section{Empty}

\subsubsection{Generation of platform-specific genotype error profiles}

To generate a genotyping error profile, \n{3} datasets were used;
(a) one \emph{truth} genotype dataset, along with a corresponding confidence map,
(b) genome-wide allele frequency information, and
(c) one \emph{typed} genotype dataset.
In each, only \glspl{snp} were considered.
First, \emph{truth} genotypes were only considered, if resulting from high-confidence variant calls (positions lie within confidence regions).


using Illumina Platinum Genomes, NA12878, as `truth' data (build hg19, version 8.0.1)
\footnote{Illumina Platinum Genomes: \url{http://www.illumina.com/platinumgenomes/}}
\footnote{\Gls{wgs} data for individual NA12878; VCF and BED files downloaded from \mbox{\url{ftp://ussd-ftp.illumina.com//hg19/8.0.1/NA12878/}}}
allele frequencies were taken from 1000G (SNPs, all autosomes) and matched to the platinum dataset
there, only \gls{snp} contained in the confident regions were retained,
and a 0 genotype was assumed at sites not present in platinum, but for which frequency information was available.

Genotyping error profiles were established for different platforms, NGS or genotyping data.
Variant sites were matched between truth and typed data based on
physical position on each chromosome and both reference and alternative alleles.
Sites present in the truth dataset, but not in the typed dataset were discarded.

Frequency-dependent genotype errors


Genotype error rates are defined as
\[
\varepsilon_{i \rightarrow j} =
\begin{bmatrix}
	\varepsilon_{0 \rightarrow 0} & \varepsilon_{0 \rightarrow 1} & \varepsilon_{0 \rightarrow 2} \\
	\varepsilon_{1 \rightarrow 0} & \varepsilon_{1 \rightarrow 1} & \varepsilon_{1 \rightarrow 2} \\
	\varepsilon_{2 \rightarrow 0} & \varepsilon_{2 \rightarrow 1} & \varepsilon_{2 \rightarrow 2}
\end{bmatrix}
=
\begin{bmatrix}
	(1 - e)^2  &  2e(1 - e)      &  e^2       \\
	e(1 - e)   &  e^2 (1 - e)^2  &  e(1 - e)  \\
	e^2 &  2e(1 - e)      &  (1 - e)^2
\end{bmatrix}
\]
where $e$ is the allelic error rate and $i, j \in \lbrace 0, 1, 2 \rbrace$





\subsection{Genotype proportions}


Expected genotype pair proportions in unrelated individuals, where $p$ and $q = 1 - p$ denote the frequency of the reference and alternative allele, respectively:

\hfill \break
\begin{minipage}{0.65\textwidth}
\begin{tabular}[t]{c|c|c|c|c|c}
\multicolumn{1}{c}{\rule[-1.5ex]{0pt}{0pt}} &
\multicolumn{1}{c}{$h_{0}, h_{0}$} &
\multicolumn{1}{c}{$h_{0}, h_{1}$} &
\multicolumn{1}{c}{$h_{1}, h_{0}$} &
\multicolumn{1}{c}{$h_{1}, h_{1}$} &
\multicolumn{1}{c}{} \\ \cline{2-5}
\rule{0pt}{3ex}\rule[-1.5ex]{0pt}{0pt}
  $h_{0}, h_{0}$    &  $p^{4}$       &  $p^{3}q$      &  $p^{3}q$      &  $p^{2}q^{2}$  &    $g_{0}$  \\ \cline{2-5}
\rule{0pt}{3ex}\rule[-1.5ex]{0pt}{0pt}
  $h_{0}, h_{1}$    &  $p^{3}q$      &  $p^{2}q^{2}$  &  $p^{2}q^{2}$  &  $pq^{3}$      &    $g_{1}$  \\ \cline{2-5}
\rule{0pt}{3ex}\rule[-1.5ex]{0pt}{0pt}
  $h_{1}, h_{0}$    &  $p^{3}q$      &  $p^{2}q^{2}$  &  $p^{2}q^{2}$  &  $pq^{3}$      &    $g_{1}$  \\ \cline{2-5}
\rule{0pt}{3ex}\rule[-1.5ex]{0pt}{0pt}
  $h_{1}, h_{1}$    &  $p^{2}q^{2}$  &  $pq^{3}$      &  $pq^{3}$      &  $q^{4}$       &    $g_{2}$  \\ \cline{2-5}
\multicolumn{1}{c}{} &
\multicolumn{1}{c}{$g_{0}$} &
\multicolumn{1}{c}{$g_{1}$} &
\multicolumn{1}{c}{$g_{1}$} &
\multicolumn{1}{c}{$g_{2}$} &
\multicolumn{1}{c}{}
\end{tabular}
\end{minipage}
\begin{minipage}{0.35\textwidth}
\begin{flushleft}
$P(g_{0}, g_{0}) = p^{4} $        \\[1.5ex]
$P(g_{0}, g_{1}) = 4p^{3}q $      \\[1.5ex]
$P(g_{0}, g_{2}) = 2p^{2}q^{2} $  \\[1.5ex]
$P(g_{1}, g_{1}) = 4p^{2}q^{2} $  \\[1.5ex]
$P(g_{1}, g_{2}) = 4pq^{3} $      \\[1.5ex]
$P(g_{2}, g_{2}) = q^{4} $
\end{flushleft}
\end{minipage}

\[
\begin{split}
	(p+q)^{4}	& = (p+q)^{2} (p+q)^{2} \\
  	& = (p^2 + 2pq + q^2) (p^2 + 2pq + q^2) \\
  	& = p^{4} + 4p^{3}q + 2p^{2}q^{2} + 4p^{2}q^{2} + 4pq^{3} + q^{4}
\end{split}
\]


Expected proportions for genotype pairs where the two individuals share a haplotype IBD (marked by $*$):

\hfill \break%
\begin{minipage}{0.65\textwidth}
\begin{tabular}[t]{c|c|c|c|c|c}
\multicolumn{1}{c}{\rule[-1.5ex]{0pt}{0pt}} &
\multicolumn{1}{c}{$h_{0}, h^*_{0}$} &
\multicolumn{1}{c}{$h_{0}, h^*_{1}$} &
\multicolumn{1}{c}{$h_{1}, h^*_{0}$} &
\multicolumn{1}{c}{$h_{1}, h^*_{1}$} &
\multicolumn{1}{c}{} \\ \cline{2-5}
\rule{0pt}{3ex}\rule[-1.5ex]{0pt}{0pt}
  $h_{0}, h^*_{0}$    &  $p^{3}$   &  $0$       &  $p^{2}q$  &  $0$       &    $g_{0}$  \\ \cline{2-5}
\rule{0pt}{3ex}\rule[-1.5ex]{0pt}{0pt}
  $h_{0}, h^*_{1}$    &  $0$       &  $p^{2}q$  &  $0$       &  $pq^{2}$  &    $g_{1}$  \\ \cline{2-5}
\rule{0pt}{3ex}\rule[-1.5ex]{0pt}{0pt}
  $h_{1}, h^*_{0}$    &  $p^{2}q$  &  $0$       &  $pq^{2}$  &  $0$       &    $g_{1}$  \\ \cline{2-5}
\rule{0pt}{3ex}\rule[-1.5ex]{0pt}{0pt}
  $h_{1}, h^*_{1}$    &  $0$       &  $pq^{2}$  &  $0$       &  $q^{3}$   &    $g_{2}$  \\ \cline{2-5}
\multicolumn{1}{c}{} &
\multicolumn{1}{c}{$g_{0}$} &
\multicolumn{1}{c}{$g_{1}$} &
\multicolumn{1}{c}{$g_{1}$} &
\multicolumn{1}{c}{$g_{2}$} &
\multicolumn{1}{c}{}
\end{tabular}
\end{minipage}
\begin{minipage}{0.35\textwidth}
\begin{flushleft}
$P_{ibd}(g_{0}, g_{0}) = p^{3} $           \\[1.5ex]
$P_{ibd}(g_{0}, g_{1}) = 2p^{2}q $         \\[1.5ex]
$P_{ibd}(g_{0}, g_{2}) = 0 $               \\[1.5ex]
$P_{ibd}(g_{1}, g_{1}) = p^{2}q + pq^{2} $ \\[1.5ex]
$P_{ibd}(g_{1}, g_{2}) = 2pq^{2} $         \\[1.5ex]
$P_{ibd}(g_{2}, g_{2}) = q^{3} $
\end{flushleft}
\end{minipage}

\[
\begin{split}
(p+q)^{3}	& = (p+q)^{2} (p+q) \\
  			& = (p^2 + 2pq + q^2) (p+q) \\
  			& = p^{3} + 2p^{2}q + pq^{2} + p^{2}q  + 2pq^{2} + q^{3}
\end{split}
\]


The order of genotypes forming a pair is ignored, as is the order of haplotypes that form a genotype. %See Thompson (1975, 1991).


\newpage


Kimura \& Ohta (1973)
\[
t(f) = \frac{-2f}{1-f}\log(f)
\]

Griffiths \& Tavar√© (1998)
\[
t(n, k) = 2 {n-1 \choose k}^{-1} \sum_{j=2}^{n} {n-j \choose k-1} \frac{n-j+1}{n(j-1)}
\]
